Failing testcase: tests/hooks/test_stdlib.py::test_tz

The testcase "test_tz" fails on a clean system installation. The problems seems to be in the `samples/tz/test_tz.py` script itself. Interestingly, reporting the time for Brazil seems to work just fine, but then it fails for US/Eastern.

Test setup on a clean Ubuntu 24.04 (server):

```
sudo apt install build-essential git python3-dev python3-venv

git clone https://github.com/marcelotduarte/cx_Freeze
cd cx_Freeze

python3 -m venv .venv
. .venv/bin/activate
pip install -r requirements-dev.txt
pip install -r tests/requirements.txt
pip install -e.
pre-commit install --install-hooks --overwrite -t pre-commit
pytest --dist=loadfile -nauto -v
```

<details>
<summary>Test output</summary>

```
================================================= test session starts ==================================================
platform linux -- Python 3.12.3, pytest-8.4.1, pluggy-1.6.0 -- /home/fklemme/Projects/fklemme/cx_Freeze/.venv/bin/python3
cachedir: .pytest_cache
rootdir: /home/fklemme/Projects/fklemme/cx_Freeze
configfile: pyproject.toml
testpaths: tests
plugins: cov-6.2.1, xdist-3.8.0, timeout-2.4.0, mock-3.14.1, anyio-4.10.0
6 workers [367 items]  skipped
scheduling tests via LoadFileScheduling

tests/test_modulefinder.py::test_finder[absolute_import_test]
tests/test_command_build_exe.py::test_build_exe_finalize_options[build-exe=none]
tests/hooks/test_multiprocessing.py::test_multiprocessing[sample0-fork]
tests/test_freezer.py::test_freezer_target_dir_empty
tests/test_executables.py::test_executables[setup_toml]
tests/test_winmsvcr.py::test_files
[gw5] [  0%] SKIPPED tests/test_winmsvcr.py::test_files
tests/test_winmsvcr.py::test_build_with_include_msvcr[False]
[gw5] [  0%] SKIPPED tests/test_winmsvcr.py::test_build_with_include_msvcr[False]
tests/test_winmsvcr.py::test_build_with_include_msvcr[True]
# ...
[gw5] [ 68%] PASSED tests/test_cli.py::test_cxfreeze
tests/test_cli.py::test_cxfreeze_help
[gw3] [ 69%] FAILED tests/hooks/test_stdlib.py::test_tz[]
tests/hooks/test_stdlib.py::test_tz[zip_packages]
[gw0] [ 69%] PASSED tests/hooks/test_multiprocess.py::test_multiprocess[sample3-spawn]
tests/hooks/test_multiprocess.py::test_multiprocess[sample3-forkserver]
[gw3] [ 69%] FAILED tests/hooks/test_stdlib.py::test_tz[zip_packages]
tests/test_command_bdist_deb.py::test_bdist_deb_not_posix
[gw3] [ 70%] SKIPPED tests/test_command_bdist_deb.py::test_bdist_deb_not_posix
tests/test_command_bdist_deb.py::test_bdist_deb_not_alien
# ...
[gw5] [100%] PASSED tests/hooks/test_scikit.py::test_sklearn[zip_packages]

======================================================= FAILURES =======================================================
______________________________________________________ test_tz[] _______________________________________________________
[gw3] linux -- Python 3.12.3 /home/fklemme/Projects/fklemme/cx_Freeze/.venv/bin/python3

tmp_package = <tests.conftest.TempPackageVenv object at 0x73ad7fbe47d0>, zip_packages = False

    @pytest.mark.venv(install_dependencies=False)
    @zip_packages
    def test_tz(tmp_package, zip_packages: bool) -> None:
        """Test if zoneinfo hook is working correctly."""
        tmp_package.create_from_sample("tz")
        if zip_packages:
            pyproject = tmp_package.path / "pyproject.toml"
            buf = pyproject.read_bytes().decode().splitlines()
            buf += ['zip_include_packages = "*"', 'zip_exclude_packages = ""']
            pyproject.write_bytes("\n".join(buf).encode("utf_8"))
        result = tmp_package.freeze()
        if "? tzdata imported from zoneinfo_hook" in str(result.stdout):
            tmp_package.install_dependencies()
            tmp_package.freeze()

        executable = tmp_package.executable("test_tz")
        assert executable.is_file()
        result = tmp_package.run(executable, timeout=TIMEOUT)
>       result.stdout.fnmatch_lines(
            [
                "TZPATH: *",
                "Available timezones: *",
                "UTC time: *",
                "Brazil time: *",
                "US Eastern time: *",
            ]
        )
E       Failed: fnmatch: 'TZPATH: *'
E          with: "TZPATH: ('/tmp/pytest-of-fklemme/pytest-2/popen-gw3/test_tz0/build/exe.linux-x86_64-3.12/share/zoneinfo',)"
E       fnmatch: 'Available timezones: *'
E          with: 'Available timezones: 501'
E       fnmatch: 'UTC time: *'
E          with: 'UTC time: Mon, 11 Aug 2025 11:35:03 +0000'
E       fnmatch: 'Brazil time: *'
E          with: 'Brazil time: Mon, 11 Aug 2025 08:35:03 -0300'
E       remains unmatched: 'US Eastern time: *'

/home/fklemme/Projects/fklemme/cx_Freeze/tests/hooks/test_stdlib.py:252: Failed
------------------------------------------------- Captured stdout call -------------------------------------------------
running build
running build_exe
Missing modules:
? tzdata imported from zoneinfo
This is not necessarily a problem - the modules may not be needed on this platform.



TZPATH: ('/tmp/pytest-of-fklemme/pytest-2/popen-gw3/test_tz0/build/exe.linux-x86_64-3.12/share/zoneinfo',)
Available timezones: 501
UTC time: Mon, 11 Aug 2025 11:35:03 +0000
Brazil time: Mon, 11 Aug 2025 08:35:03 -0300

Traceback (most recent call last):
  File "/usr/lib/python3.12/zoneinfo/_common.py", line 12, in load_tzdata
    return resources.files(package_name).joinpath(resource_name).open("rb")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/importlib/resources/_common.py", line 46, in wrapper
    return func(anchor)
           ^^^^^^^^^^^^
  File "/usr/lib/python3.12/importlib/resources/_common.py", line 56, in files
    return from_package(resolve(anchor))
                        ^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/functools.py", line 909, in wrapper
    return dispatch(args[0].__class__)(*args, **kw)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/importlib/resources/_common.py", line 82, in _
    return importlib.import_module(cand)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1310, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1310, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1324, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'tzdata'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/fklemme/Projects/fklemme/cx_Freeze/cx_Freeze/initscripts/__startup__.py", line 133, in run
    module_init.run(f"__main__{name}")
  File "/home/fklemme/Projects/fklemme/cx_Freeze/cx_Freeze/initscripts/console.py", line 25, in run
    exec(code, main_globals)
  File "test_tz.py", line 18, in <module>
    tz2 = ZoneInfo("US/Eastern")
          ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/zoneinfo/_common.py", line 24, in load_tzdata
    raise ZoneInfoNotFoundError(f"No time zone found with key {key}")
zoneinfo._common.ZoneInfoNotFoundError: 'No time zone found with key US/Eastern'

________________________________________________ test_tz[zip_packages] _________________________________________________
[gw3] linux -- Python 3.12.3 /home/fklemme/Projects/fklemme/cx_Freeze/.venv/bin/python3

tmp_package = <tests.conftest.TempPackageVenv object at 0x73ad7fc4b830>, zip_packages = True

    @pytest.mark.venv(install_dependencies=False)
    @zip_packages
    def test_tz(tmp_package, zip_packages: bool) -> None:
        """Test if zoneinfo hook is working correctly."""
        tmp_package.create_from_sample("tz")
        if zip_packages:
            pyproject = tmp_package.path / "pyproject.toml"
            buf = pyproject.read_bytes().decode().splitlines()
            buf += ['zip_include_packages = "*"', 'zip_exclude_packages = ""']
            pyproject.write_bytes("\n".join(buf).encode("utf_8"))
        result = tmp_package.freeze()
        if "? tzdata imported from zoneinfo_hook" in str(result.stdout):
            tmp_package.install_dependencies()
            tmp_package.freeze()

        executable = tmp_package.executable("test_tz")
        assert executable.is_file()
        result = tmp_package.run(executable, timeout=TIMEOUT)
>       result.stdout.fnmatch_lines(
            [
                "TZPATH: *",
                "Available timezones: *",
                "UTC time: *",
                "Brazil time: *",
                "US Eastern time: *",
            ]
        )
E       Failed: fnmatch: 'TZPATH: *'
E          with: "TZPATH: ('/tmp/pytest-of-fklemme/pytest-2/popen-gw3/test_tz1/build/exe.linux-x86_64-3.12/share/zoneinfo',)"
E       fnmatch: 'Available timezones: *'
E          with: 'Available timezones: 501'
E       fnmatch: 'UTC time: *'
E          with: 'UTC time: Mon, 11 Aug 2025 11:35:13 +0000'
E       fnmatch: 'Brazil time: *'
E          with: 'Brazil time: Mon, 11 Aug 2025 08:35:13 -0300'
E       remains unmatched: 'US Eastern time: *'

/home/fklemme/Projects/fklemme/cx_Freeze/tests/hooks/test_stdlib.py:252: Failed
------------------------------------------------- Captured stdout call -------------------------------------------------
running build
running build_exe
Missing modules:
? tzdata imported from zoneinfo
This is not necessarily a problem - the modules may not be needed on this platform.



TZPATH: ('/tmp/pytest-of-fklemme/pytest-2/popen-gw3/test_tz1/build/exe.linux-x86_64-3.12/share/zoneinfo',)
Available timezones: 501
UTC time: Mon, 11 Aug 2025 11:35:13 +0000
Brazil time: Mon, 11 Aug 2025 08:35:13 -0300

Traceback (most recent call last):
  File "/usr/lib/python3.12/zoneinfo/_common.py", line 12, in load_tzdata
    return resources.files(package_name).joinpath(resource_name).open("rb")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/importlib/resources/_common.py", line 46, in wrapper
    return func(anchor)
           ^^^^^^^^^^^^
  File "/usr/lib/python3.12/importlib/resources/_common.py", line 56, in files
    return from_package(resolve(anchor))
                        ^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/functools.py", line 909, in wrapper
    return dispatch(args[0].__class__)(*args, **kw)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/importlib/resources/_common.py", line 82, in _
    return importlib.import_module(cand)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1310, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1310, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1324, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'tzdata'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/fklemme/Projects/fklemme/cx_Freeze/cx_Freeze/initscripts/__startup__.py", line 133, in run
    module_init.run(f"__main__{name}")
  File "/home/fklemme/Projects/fklemme/cx_Freeze/cx_Freeze/initscripts/console.py", line 25, in run
    exec(code, main_globals)
  File "test_tz.py", line 18, in <module>
    tz2 = ZoneInfo("US/Eastern")
          ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/zoneinfo/_common.py", line 24, in load_tzdata
    raise ZoneInfoNotFoundError(f"No time zone found with key {key}")
zoneinfo._common.ZoneInfoNotFoundError: 'No time zone found with key US/Eastern'

=============================================== short test summary info ================================================
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[build-exe=none]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[build-exe=dist]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[excludes='tkinter,unittest']
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[excludes=['tkinter','unittest']]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[include-msvcr=none]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[include-msvcr=false]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[include-msvcr=true]
PASSED tests/test_modulefinder.py::test_finder[absolute_import_test]
PASSED tests/test_modulefinder.py::test_finder[coding_default_utf8_test]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[replace_paths=*]
PASSED tests/test_modulefinder.py::test_finder[coding_explicit_cp1252_test]
PASSED tests/test_modulefinder.py::test_finder[coding_explicit_utf8_test]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[replace_paths=*=]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[replace_paths=[*=]]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[silent=none->0]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[silent=false->0]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[silent=true->1]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[silent-level=none->0]
PASSED tests/test_modulefinder.py::test_finder[extended_opargs_test]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[silent-level=0->0]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[silent-level=1->1]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[silent-level=2->2]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[silent-level=3->3]
PASSED tests/test_modulefinder.py::test_finder[import_call_test]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[zip_include_packages/zip_exclude_packages=none/none]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[zip_include_package=[*]]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[zip_exclude_packages=*]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[zip_include_packages/zip_exclude_packages=namespace/namespace]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[zip_include_package=*]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[zip_filename=none]
PASSED tests/test_modulefinder.py::test_finder[maybe_test]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[zip_filename=test]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[zip_filename=test.zip]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[zip_filename=0]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[zip_filename=1]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[no_compress=false]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options[no_compress=true]
PASSED tests/test_command_build_exe.py::test_build_exe_finalize_options_raises[build-exe=build]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--build-exe(notused)]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--build-exe=]
PASSED tests/test_modulefinder.py::test_finder[maybe_test_new]
PASSED tests/test_modulefinder.py::test_finder[namespace_test]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--build-exe=dist]
PASSED tests/test_modulefinder.py::test_finder[nested_namespace_test]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--excludes=]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--excludes=tkinter,unittest]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--includes=]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--includes=tkinter,unittest]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--packages=]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--packages=tkinter,unittest]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--replace-paths=*=]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--bin-excludes=]
PASSED tests/test_modulefinder.py::test_finder[package_test]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--bin-includes=]
PASSED tests/test_modulefinder.py::test_finder[relative_import_test]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--bin-path-excludes=]
PASSED tests/test_modulefinder.py::test_finder[relative_import_test_2]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--bin-path-includes=]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--include-files=]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--zip-includes=]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--zip-include-packages/--zip-exclude-packages(notused)]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--zip-include-packages=/--zip-exclude-packages=]
PASSED tests/test_modulefinder.py::test_finder[relative_import_test_3]
PASSED tests/test_modulefinder.py::test_finder[relative_import_test_4]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--zip-include-package=*/--zip-exclude-packages=]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--zip-include-packages=/--zip-exclude-packages=*]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--zip-include-packages/--zip-exclude-packages=namespace/namespace]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--zip-filename=]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--zip-filename=test]
PASSED tests/test_modulefinder.py::test_finder[same_name_as_bad_test]
PASSED tests/test_modulefinder.py::test_finder[sub_package_test]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--zip-filename=test.zip]
PASSED tests/test_modulefinder.py::test_bytecode
PASSED tests/test_modulefinder.py::test_zip_include_packages
PASSED tests/test_modulefinder.py::test_zip_exclude_packages
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--no-compress(notused),--zip-filename(notused)]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--no-compress]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--optimize(notused)]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--optimize=0]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--optimize=1]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--optimize=2]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--optimize(-O0]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--optimize(-O1]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--optimize(-O2]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--silent(notused)]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--silent]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--silent-level=0->0]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--silent-level=1->1]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--silent-level=2->2]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--silent-level=3->3]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--include-msvcr(notused)]
PASSED tests/test_command_build_exe.py::test_build_exe_script_args[--include-msvcr]
PASSED tests/test_freezer.py::test_freezer_target_dir_empty
PASSED tests/test_executables.py::test_executables[setup_toml]
PASSED tests/test_freezer.py::test_freezer_target_dir_dist
PASSED tests/test_freezer.py::test_freezer_target_dir_utf8
PASSED tests/test_freezer.py::test_freezer_target_dir_in_path
PASSED tests/test_freezer.py::test_freezer_target_dir_locked
PASSED tests/hooks/test_multiprocessing.py::test_multiprocessing[sample0-fork]
PASSED tests/test_freezer.py::test_freezer_default_bin_includes
PASSED tests/test_freezer.py::test_freezer_populate_zip_options_invalid_values
PASSED tests/test_modulefinder.py::test_editable_packages
PASSED tests/test_freezer.py::test_freezer_options[compress=none]
PASSED tests/test_winmsvcr.py::test_versions[15-win32-False]
PASSED tests/test_freezer.py::test_freezer_options[compress=false]
PASSED tests/hooks/test_stdlib.py::test_ctypes[]
PASSED tests/test_winmsvcr.py::test_versions[15-win-amd64-False]
PASSED tests/test_freezer.py::test_freezer_options[compress=true]
PASSED tests/test_freezer.py::test_freezer_options[excludes=['tkinter','unittest']]
PASSED tests/hooks/test_multiprocessing.py::test_multiprocessing[sample0-fork-zip_packages]
PASSED tests/test_freezer.py::test_freezer_options[include_msvcr=none]
PASSED tests/test_freezer.py::test_freezer_options[include_msvcr=false]
PASSED tests/test_executables.py::test_executables[setup_py]
PASSED tests/test_freezer.py::test_freezer_options[include_msvcr=true]
PASSED tests/test_freezer.py::test_freezer_options[replace_paths=*]
PASSED tests/test_command_build_exe.py::test_build_exe_advanced
PASSED tests/test_winmsvcr.py::test_versions[15-win-arm64-False]
PASSED tests/test_freezer.py::test_freezer_options[replace_paths=[*=]]
PASSED tests/test_freezer.py::test_freezer_options[optimize=none->0]
PASSED tests/test_command_build_exe.py::test_build_exe_asmodule
PASSED tests/test_freezer.py::test_freezer_options[optimize=0->0]
PASSED tests/test_executables.py::test_executables[setup_cfg]
PASSED tests/test_freezer.py::test_freezer_options[optimize=1->1]
PASSED tests/test_freezer.py::test_freezer_options[optimize=2->2]
PASSED tests/test_executables.py::test_executables[setup_mix]
PASSED tests/test_freezer.py::test_freezer_options[silent=none->0]
PASSED tests/test_freezer.py::test_freezer_options[silent=false->0]
PASSED tests/test_command_build_exe.py::test_zip_include_packages
PASSED tests/test_winmsvcr.py::test_versions[16-win32-False]
PASSED tests/test_freezer.py::test_freezer_options[silent=true->1]
PASSED tests/test_winmsvcr.py::test_versions[16-win-amd64-False]
PASSED tests/test_freezer.py::test_freezer_options[silent=0->0]
PASSED tests/hooks/test_stdlib.py::test_ctypes[zip_packages]
PASSED tests/test_freezer.py::test_freezer_options[silent=1->1]
PASSED tests/test_winmsvcr.py::test_versions[16-win-arm64-False]
PASSED tests/test_freezer.py::test_freezer_options[silent=2->2]
PASSED tests/hooks/test_multiprocessing.py::test_multiprocessing[sample1-fork]
PASSED tests/test_freezer.py::test_freezer_options[silent=3->3]
PASSED tests/test_winmsvcr.py::test_versions[17-win32-False]
PASSED tests/test_freezer.py::test_freezer_options[zip_include_packages/zip_exclude_packages=none/none]
PASSED tests/test_freezer.py::test_freezer_options[zip_include_package=*]
PASSED tests/hooks/test_multiprocessing.py::test_multiprocessing[sample1-fork-zip_packages]
PASSED tests/test_freezer.py::test_freezer_options[zip_exclude_packages=*]
PASSED tests/hooks/test_stdlib.py::test_sqlite[]
PASSED tests/test_freezer.py::test_freezer_options[zip_include_packages/zip_exclude_packages=namespace/namespace]
PASSED tests/test_freezer.py::test_freezer_zip_filename[zip_filename=none]
PASSED tests/hooks/test_multiprocessing.py::test_multiprocessing[sample2-fork]
PASSED tests/test_freezer.py::test_freezer_zip_filename[zip_filename=test]
PASSED tests/test_freezer.py::test_freezer_zip_filename[zip_filename=test.zip]
PASSED tests/test_executables.py::test_executables[setup_adv_toml]
PASSED tests/test_winmsvcr.py::test_versions[17-win-amd64-True]
PASSED tests/test_freezer.py::test_freezer_zip_filename[zip_filename=test.zip/target_dir=utf_8/portuguese]
PASSED tests/test_command_build_exe.py::test_zip_exclude_packages
PASSED tests/test_command_bdist_appimage.py::test_bdist_appimage_download_appimagetool
PASSED tests/test_freezer.py::test_freezer_zip_filename[zip_filename=test.zip/target_dir=utf_8/chinese]
PASSED tests/test_command_bdist_appimage.py::test_bdist_appimage_download_runtime
PASSED tests/test_command_bdist_appimage.py::test_bdist_appimage_target_name
PASSED tests/test_command_bdist_appimage.py::test_bdist_appimage_target_name_and_version
PASSED tests/test_command_bdist_appimage.py::test_bdist_appimage_target_name_and_name_none
PASSED tests/test_command_bdist_appimage.py::test_bdist_appimage_target_name_and_version_none
PASSED tests/test_freezer.py::test_freezer_zip_filename[zip_filename=none/compress=true]
PASSED tests/hooks/test_stdlib.py::test_sqlite[zip_packages]
PASSED tests/test_freezer.py::test_freezer_zip_filename[zip_filename=none/compress=false]
PASSED tests/test_command_bdist_appimage.py::test_bdist_appimage_target_name_with_extension
PASSED tests/test_freezer.py::test_freezer_zip_filename[zip_filename=name/compress=false]
PASSED tests/test_command_bdist_appimage.py::test_bdist_appimage_skip_build
PASSED tests/test_command_bdist_appimage.py::test_bdist_appimage_skip_build_after_build_exe
PASSED tests/test_executables.py::test_executables[setup_adv_py]
PASSED tests/test_executables.py::test_valid[base-console-console-cpython-312-x86_64-linux-gnu]
PASSED tests/test_executables.py::test_valid[base-absolutepath-console_test]
PASSED tests/test_executables.py::test_valid[init_script-none-console.py]
PASSED tests/test_executables.py::test_valid[init_script-console-console.py]
PASSED tests/test_executables.py::test_valid[init_script-absolutepath-console_test.py]
PASSED tests/test_executables.py::test_valid[target_name-none-test]
PASSED tests/test_executables.py::test_valid[target_name-test1-test1]
PASSED tests/test_executables.py::test_valid[target_name-12345-12345]
PASSED tests/test_executables.py::test_valid[target_name-test-0.1-test-0.1]
PASSED tests/test_executables.py::test_valid[target_name-test.exe-test.exe]
PASSED tests/test_executables.py::test_valid[icon-icon-multiple]
PASSED tests/test_executables.py::test_valid[base-none-console_legacy-]
PASSED tests/test_executables.py::test_valid[base-console_legacy-console_legacy-cpython-312-x86_64-linux-gnu]
PASSED tests/test_executables.py::test_valid[base-gui-console-cpython-312-x86_64-linux-gnu]
PASSED tests/test_executables.py::test_valid[base-service-console-cpython-312-x86_64-linux-gnu]
PASSED tests/test_executables.py::test_invalid[executables-invalid-empty]
PASSED tests/test_executables.py::test_invalid[executables-invalid-string]
PASSED tests/test_executables.py::test_invalid[executable-invalid-base]
PASSED tests/test_executables.py::test_invalid[executable-invalid-init_script]
PASSED tests/test_executables.py::test_invalid[executable-invalid-target_name-with-path]
PASSED tests/hooks/test_numpy.py::test_matplotlib[]
PASSED tests/hooks/test_stdlib.py::test_sqlite_ext[]
PASSED tests/test_command_bdist_appimage.py::test_bdist_appimage_simple
PASSED tests/hooks/test_multiprocess.py::test_multiprocess[sample0-fork]
PASSED tests/test_executables.py::test_valid_icon
PASSED tests/test_winmsvcr.py::test_versions[17-win-arm64-False]
PASSED tests/hooks/test_multiprocess.py::test_multiprocess[sample1-fork]
PASSED tests/hooks/test_numpy.py::test_matplotlib[zip_packages]
PASSED tests/hooks/test_stdlib.py::test_sqlite_ext[zip_packages]
PASSED tests/hooks/test_multiprocess.py::test_multiprocess[sample2-fork]
PASSED tests/hooks/test_multiprocessing.py::test_multiprocessing[sample2-fork-zip_packages]
PASSED tests/test_winmsvcr.py::test_invalid[17-win-amd64-RuntimeError-Version is not expected]
PASSED tests/test_winmsvcr.py::test_invalid[18-win-amd64-RuntimeError-Version is not expected]
PASSED tests/test_winmsvcr.py::test_invalid[17--RuntimeError-Architecture not supported]
PASSED tests/hooks/test_multiprocess.py::test_multiprocess[sample3-fork]
PASSED tests/hooks/test_stdlib.py::test_ssl[]
PASSED tests/test_winmsvcr.py::test_invalid[17-win64-RuntimeError-Architecture not supported]
PASSED tests/hooks/test_multiprocess.py::test_multiprocess[sample0-spawn]
PASSED tests/test_winmsvcr.py::test_repack_main
PASSED tests/hooks/test_multiprocess.py::test_multiprocess[sample1-spawn]
PASSED tests/test_executables.py::test_not_found_icon
PASSED tests/hooks/test_multiprocess.py::test_multiprocess[sample2-spawn]
PASSED tests/hooks/test_multiprocessing.py::test_multiprocessing[sample3-fork]
PASSED tests/test_winmsvcr.py::test_repack_main_no_option
PASSED tests/hooks/test_stdlib.py::test_ssl[zip_packages]
PASSED tests/test_cli.py::test_cxfreeze
PASSED tests/hooks/test_multiprocess.py::test_multiprocess[sample3-spawn]
PASSED tests/test_command_bdist_deb.py::test_bdist_deb_not_alien
PASSED tests/test_command_bdist_deb.py::test_bdist_deb_not_fakeroot
PASSED tests/test_command_bdist_deb.py::test_bdist_deb_dry_run
PASSED tests/hooks/test_numpy.py::test_pandas[]
PASSED tests/hooks/test_multiprocessing.py::test_multiprocessing[sample3-fork-zip_packages]
PASSED tests/test_executables.py::test_executable_rename
PASSED tests/test_cli.py::test_cxfreeze_help
PASSED tests/test_cli.py::test_cxfreeze_additional_help
PASSED tests/hooks/test_multiprocess.py::test_multiprocess[sample3-forkserver]
PASSED tests/hooks/test_multiprocessing.py::test_multiprocessing[sample0-spawn]
PASSED tests/hooks/test_crypto.py::test_argon2[]
PASSED tests/hooks/test_crypto.py::test_argon2[zip_packages]
PASSED tests/test_cli.py::test_cxfreeze_debug_verbose
PASSED tests/test_command_bdist_rpm.py::test_bdist_rpm_not_rpmbuild
PASSED tests/hooks/test_crypto.py::test_bcrypt[]
PASSED tests/hooks/test_crypto.py::test_bcrypt[zip_packages]
PASSED tests/test_executables.py::test_executable_namespace[namespace_package]
PASSED tests/hooks/test_crypto.py::test_crypto[]
PASSED tests/test_cli.py::test_cxfreeze_target_name_not_isidentifier
PASSED tests/hooks/test_crypto.py::test_crypto[zip_packages]
PASSED tests/test_module.py::test_implicit_namespace_package
PASSED tests/hooks/test_crypto.py::test_cryptography[]
PASSED tests/test_cli.py::test_cxfreeze_deprecated_behavior
PASSED tests/hooks/test_crypto.py::test_cryptography[zip_packages]
PASSED tests/hooks/test_numpy.py::test_pandas[zip_packages]
PASSED tests/test_executables.py::test_executable_namespace[namespace_package_zip_packages]
PASSED tests/hooks/test_multiprocessing.py::test_multiprocessing[sample0-spawn-zip_packages]
PASSED tests/hooks/test_numpy.py::test_rasterio[]
PASSED tests/hooks/test_ortools.py::test_ortools[]
PASSED tests/test_cli.py::test_cxfreeze_deprecated_option
PASSED tests/test_module.py::test_build_constants[]
PASSED tests/test_cli.py::test_cxfreeze_without_options
PASSED tests/hooks/test_numpy.py::test_rasterio[zip_packages]
PASSED tests/test_cli.py::test_cxfreeze_include_path
PASSED tests/hooks/test_ortools.py::test_ortools[zip_packages]
PASSED tests/test___init__.py::test_exposed_namespaces[None-extra_modules0]
PASSED tests/test___init__.py::test_exposed_namespaces[win32-extra_modules1]
PASSED tests/test___init__.py::test_exposed_namespaces[darwin-extra_modules2]
PASSED tests/test___init__.py::test_exposed_namespaces[linux-extra_modules3]
PASSED tests/hooks/test_numpy.py::test_shapely[]
PASSED tests/test_module.py::test_build_constants[zip_packages]
PASSED tests/test_module.py::test_invalid[invalid-constant-isidentifier]
PASSED tests/test_module.py::test_invalid[invalid-constant-iskeyword]
PASSED tests/hooks/test_numpy.py::test_shapely[zip_packages]
PASSED tests/test_dep_parser.py::test_parser[patchelf-bind]
PASSED tests/test_dep_parser.py::test_elf_parser
PASSED tests/test_dep_parser.py::test_verify_patchelf
PASSED tests/test_dep_parser.py::test_verify_patchelf_older
PASSED tests/hooks/test_multiprocessing.py::test_multiprocessing[sample1-spawn]
PASSED tests/hooks/test_av.py::test_av[]
PASSED tests/test_executables.py::test_executable_namespace[nested_namespace_package]
PASSED tests/hooks/test_av.py::test_av[zip_packages]
PASSED tests/test_module.py::test_egg_info
PASSED tests/test_finder.py::TestModuleFinder::test_scan_code
PASSED tests/test_finder.py::TestModuleFinder::test_not_import_invalid_module_name
PASSED tests/test_finder.py::TestModuleFinder::test_invalid_syntax
PASSED tests/hooks/test_scikit.py::test_scipy[]
PASSED tests/test_finder.py::TestModuleFinder::test_find_spec
PASSED tests/hooks/test_pyarrow.py::test_pyarrow[]
PASSED tests/hooks/test_pyproj.py::test_pyproj[]
PASSED tests/hooks/test_multiprocessing.py::test_multiprocessing[sample1-spawn-zip_packages]
PASSED tests/hooks/test_pyproj.py::test_pyproj[zip_packages]
PASSED tests/hooks/test_pyarrow.py::test_pyarrow[zip_packages]
PASSED tests/hooks/test_numpy.py::test_vtk[]
PASSED tests/hooks/test_pymupdf.py::test_pymupdf[]
PASSED tests/hooks/test_pytz.py::test_pytz[]
PASSED tests/test_executables.py::test_executable_namespace[nested_namespace_package_zip_packages]
PASSED tests/hooks/test_pytz.py::test_pytz[zip_packages]
PASSED tests/hooks/test_pymupdf.py::test_pymupdf[zip_packages]
PASSED tests/hooks/test_numpy.py::test_vtk[zip_packages]
PASSED tests/hooks/test_multiprocessing.py::test_multiprocessing[sample2-spawn]
PASSED tests/test_executables.py::test_valid_sys_path
PASSED tests/hooks/test_pyzmq.py::test_pyzmq[]
PASSED tests/hooks/test_zeroconf.py::test_zeroconf[]
PASSED tests/hooks/test_pillow.py::test_pillow[]
PASSED tests/hooks/test_async.py::test_anyio[]
PASSED tests/hooks/test_pyzmq.py::test_pyzmq[zip_packages]
PASSED tests/hooks/test_zeroconf.py::test_zeroconf[zip_packages]
PASSED tests/hooks/test_multiprocessing.py::test_multiprocessing[sample2-spawn-zip_packages]
PASSED tests/hooks/test_pillow.py::test_pillow[zip_packages]
PASSED tests/hooks/test_async.py::test_anyio[zip_packages]
PASSED tests/test_command_install_exe.py::test_install_exe_finalize_options[install-dir=none]
PASSED tests/test_command_install_exe.py::test_install_exe_finalize_options[install-dir=dist]
PASSED tests/hooks/test_scikit.py::test_scipy[zip_packages]
PASSED tests/test_command_install.py::test_install
PASSED tests/hooks/test_multiprocessing.py::test_multiprocessing[sample3-spawn]
PASSED tests/test___main__.py::test___main__
PASSED tests/hooks/test_multiprocessing.py::test_multiprocessing[sample3-spawn-zip_packages]
PASSED tests/test_command_build_exe_excludes.py::test_build_exe_excludes
PASSED tests/test_command_install.py::test_install_pyproject
PASSED tests/hooks/test_multiprocessing.py::test_multiprocessing[sample3-forkserver]
PASSED tests/test_command_build.py::test_build
PASSED tests/test_command_build.py::test_build_raises[build-exe=]
PASSED tests/hooks/test_multiprocessing.py::test_multiprocessing[sample3-forkserver-zip_packages]
PASSED tests/hooks/test_scikit.py::test_sklearn[]
PASSED tests/hooks/test_scikit.py::test_sklearn[zip_packages]
FAILED tests/hooks/test_stdlib.py::test_tz[] - Failed: fnmatch: 'TZPATH: *'
FAILED tests/hooks/test_stdlib.py::test_tz[zip_packages] - Failed: fnmatch: 'TZPATH: *'
SKIPPED [1] tests/hooks/test_win32com.py:12: Windows tests
SKIPPED [1] tests/test_command_bdist_dmg.py:9: macOS tests
SKIPPED [1] tests/test_command_bdist_mac.py:14: macOS tests
SKIPPED [1] tests/test_windows_manifest.py:15: Windows tests
SKIPPED [1] tests/test_winmsvcr.py:52: Windows tests
SKIPPED [5] tests/test_winmsvcr.py:61: Windows tests
SKIPPED [1] tests/test_winversioninfo.py:32: Windows tests
SKIPPED [1] tests/test_winversioninfo.py:51: Windows tests
SKIPPED [1] tests/test_winversioninfo.py:95: Windows tests
SKIPPED [9] tests/test_winversioninfo.py:103: Windows tests
SKIPPED [1] tests/test_winversioninfo.py:125: Windows tests
SKIPPED [3] tests/test_winversioninfo.py:131: Windows tests
SKIPPED [1] tests/test_winversioninfo.py:158: Windows tests
SKIPPED [1] tests/test_winversioninfo.py:163: Windows tests
SKIPPED [1] tests/test_command_bdist_appimage.py:26: Test not on Linux platform
SKIPPED [1] tests/test_command_bdist_msi.py:27: Windows test
SKIPPED [1] tests/test_command_bdist_msi.py:38: Windows test
SKIPPED [1] tests/test_command_bdist_msi.py:50: Windows test
SKIPPED [1] tests/test_command_bdist_msi.py:60: Windows test
SKIPPED [1] tests/test_command_bdist_msi.py:72: Windows test
SKIPPED [1] tests/test_command_bdist_msi.py:83: Windows test
SKIPPED [1] tests/test_command_bdist_msi.py:95: Windows test
SKIPPED [1] tests/test_command_bdist_msi.py:106: Windows test
SKIPPED [1] tests/test_command_bdist_msi.py:117: Windows test
SKIPPED [1] tests/test_command_bdist_msi.py:151: Windows test
SKIPPED [1] tests/test_executables.py:439: Windows tests
SKIPPED [2] tests/hooks/test_stdlib.py:192: Depends on extra package: tkinter
SKIPPED [1] tests/hooks/test_stdlib.py:211: macOS test
SKIPPED [1] tests/test_command_bdist_deb.py:29: Test not on Linux platform
SKIPPED [1] tests/test_command_bdist_rpm.py:26: Test not on Linux platform
SKIPPED [2] tests/hooks/test_ortools.py:154: conda-forge test only
SKIPPED [2] tests/hooks/test_mkl.py:19: Test only on Linux64 and Python 3.10
SKIPPED [1] tests/test_plist_items.py:57: macOS test
XFAIL tests/test_command_bdist_deb.py::test_bdist_deb_simple - alien not installed
XFAIL tests/test_command_bdist_deb.py::test_bdist_deb_simple_pyproject - alien not installed
XFAIL tests/test_command_bdist_deb.py::test_bdist_deb - alien not installed
XFAIL tests/test_command_bdist_rpm.py::test_bdist_rpm_options[spec_only] - rpmbuild is not installed
XFAIL tests/test_command_bdist_rpm.py::test_bdist_rpm_options_run[spec_only] - rpmbuild is not installed
XFAIL tests/test_command_bdist_rpm.py::test_bdist_rpm_simple - rpmbuild is not installed
XFAIL tests/test_command_bdist_rpm.py::test_bdist_rpm_simple_pyproject - rpmbuild is not installed
XFAIL tests/test_command_bdist_rpm.py::test_bdist_rpm - rpmbuild is not installed
========================== 2 failed, 311 passed, 50 skipped, 8 xfailed in 1193.12s (0:19:53) ===========================
```

</details>

Manual test:

```
$ python --version
Python 3.12.3
$ python samples/tz/test_tz.py
TZPATH: ('/usr/share/zoneinfo', '/usr/lib/zoneinfo', '/usr/share/lib/zoneinfo', '/etc/zoneinfo')
Available timezones: 498
UTC time: Mon, 11 Aug 2025 12:20:13 +0000
Brazil time: Mon, 11 Aug 2025 09:20:13 -0300
Traceback (most recent call last):
  File "/usr/lib/python3.12/zoneinfo/_common.py", line 12, in load_tzdata
    return resources.files(package_name).joinpath(resource_name).open("rb")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/importlib/resources/_common.py", line 46, in wrapper
    return func(anchor)
           ^^^^^^^^^^^^
  File "/usr/lib/python3.12/importlib/resources/_common.py", line 56, in files
    return from_package(resolve(anchor))
                        ^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/functools.py", line 909, in wrapper
    return dispatch(args[0].__class__)(*args, **kw)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/importlib/resources/_common.py", line 82, in _
    return importlib.import_module(cand)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1310, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1310, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1324, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'tzdata'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/fklemme/Projects/fklemme/cx_Freeze/samples/tz/test_tz.py", line 18, in <module>
    tz2 = ZoneInfo("US/Eastern")
          ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/zoneinfo/_common.py", line 24, in load_tzdata
    raise ZoneInfoNotFoundError(f"No time zone found with key {key}")
zoneinfo._common.ZoneInfoNotFoundError: 'No time zone found with key US/Eastern'
```

--- a/samples/tz/pyproject.toml
+++ b/samples/tz/pyproject.toml
@@ -2,7 +2,7 @@
 name = "test_tz"
 version = "0.1.2.3"
 description = "cx_Freeze script to test zoneinfo/tzdata"
-dependencies = ["tzdata;sys_platform == 'win32'"]
+dependencies = ["tzdata"]
 
 [tool.cxfreeze]
 executables = ["test_tz.py"]

--- a/tests/hooks/test_stdlib.py
+++ b/tests/hooks/test_stdlib.py
@@ -6,7 +6,7 @@ import os
 
 import pytest
 
-from cx_Freeze._compat import IS_MACOS
+from cx_Freeze._compat import IS_MACOS, IS_WINDOWS
 
 TIMEOUT = 10
 
@@ -232,9 +232,10 @@ def test_tkinter_bdist_mac(tmp_package) -> None:
 
 
 @pytest.mark.venv(install_dependencies=False)
+@pytest.mark.skipif(IS_WINDOWS, reason="Windows doesn't have system timezone")
 @zip_packages
-def test_tz(tmp_package, zip_packages: bool) -> None:
-    """Test if zoneinfo hook is working correctly."""
+def test_zoneinfo(tmp_package, zip_packages: bool) -> None:
+    """Test if zoneinfo hook with system timezone is working correctly."""
     tmp_package.create_from_sample("tz")
     if zip_packages:
         pyproject = tmp_package.path / "pyproject.toml"
@@ -242,9 +243,36 @@ def test_tz(tmp_package, zip_packages: bool) -> None:
         buf += ['zip_include_packages = "*"', 'zip_exclude_packages = ""']
         pyproject.write_bytes("\n".join(buf).encode("utf_8"))
     result = tmp_package.freeze()
-    if "? tzdata imported from zoneinfo_hook" in str(result.stdout):
-        tmp_package.install_dependencies()
-        tmp_package.freeze()
+
+    executable = tmp_package.executable("test_tz")
+    assert executable.is_file()
+    result = tmp_package.run(executable, timeout=TIMEOUT)
+    if result.stderr.lines:
+        python_result = tmp_package.run("python test_tz.py", timeout=TIMEOUT)
+        if python_result.stderr.lines:
+            pytest.xfail("system timezone is broken")
+    result.stdout.fnmatch_lines(
+        [
+            "TZPATH: *",
+            "Available timezones: *",
+            "UTC time: *",
+            "Brazil time: *",
+            "US Eastern time: *",
+        ]
+    )
+
+
+@pytest.mark.venv
+@zip_packages
+def test_zoneinfo_and_tzdata(tmp_package, zip_packages: bool) -> None:
+    """Test if zoneinfo and tzdata hook is working correctly."""
+    tmp_package.create_from_sample("tz")
+    if zip_packages:
+        pyproject = tmp_package.path / "pyproject.toml"
+        buf = pyproject.read_bytes().decode().splitlines()
+        buf += ['zip_include_packages = "*"', 'zip_exclude_packages = ""']
+        pyproject.write_bytes("\n".join(buf).encode("utf_8"))
+    result = tmp_package.freeze()
 
     executable = tmp_package.executable("test_tz")
     assert executable.is_file()

